lib/staging/slac/fsm/ev/src/states/sounding.cpp

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 - 2023 Pionix GmbH and Contributors to EVerest
+#include <slac/fsm/ev/states/sounding.hpp>
+
+#include <algorithm>
+#include <cstring>
+#include <optional>
+#include <random>
+
+#include "timing_helper.hpp"
+
+#include <slac/fsm/ev/states/others.hpp>
+
+namespace slac::fsm::ev {
+
+SoundingState::SoundingState(Context& ctx, SessionParamaters session_parameters_) :
+    FSMSimpleState(ctx), session_parameters(std::move(session_parameters_)) {
+}
+
+void SoundingState::enter() {
+    ctx.signal_state("SOUNDING");
```

Comment:
"SOUNDING" is not a valid state enum. Are the correct states (MATCHED etc) also published somewhere?

Reply:
Right, this should be corrected.  I will have a look at the interface definitions.

Reply:
Added the correct state and dlink_ready publishes